app/components/postForm.tsx

```diff
@@ -0,0 +1,39 @@
+import { useFetcher } from "@remix-run/react";
+import { useEffect, useState } from "react";
+import { action } from "~/routes/api.notes.create";
+
+export const PostForm = () => {
+  const [content, setContent] = useState<string>("");
+  const [errorMessage, setErrorMessage] = useState<string>("");
+  const fetcher = useFetcher<typeof action>();
+
+  useEffect(() => {
+    if (!fetcher.data) return;
+
+    if (fetcher.state === "loading") {
+      if ("error" in fetcher.data) {
+        return setErrorMessage(fetcher.data.error);
```

Comment:
```suggestion
        setErrorMessage(fetcher.data.error);
        return;
```
戻り値型が `void` であるとはいえど, `useEffect` での戻り値はクリーンアップ処理の関数を書くものなので意味論的によくないと思います. 分けて書いてほしいです.